app/src/main/java/com/teamagam/gimelgimel/app/message/viewModel/adapter/MessageViewHolderBindVisitor.java

```diff
@@ -36,76 +31,63 @@
       GoToLocationMapInteractorFactory goToLocationMapInteractorFactory,
-      ToggleMessageOnMapInteractorFactory toggleMessageOnMapInteractorFactory, Navigator navigator,
-      GlideLoader glideLoader, MessagePresentation presentation) {
+      ToggleMessageOnMapInteractorFactory toggleMessageOnMapInteractorFactory,
+      Navigator navigator,
+      GlideLoader glideLoader,
+      MessagePresentation presentation) {
     mMessageViewHolder = messageViewHolder;
     mGoToLocationMapInteractorFactory = goToLocationMapInteractorFactory;
     mToggleMessageOnMapInteractorFactory = toggleMessageOnMapInteractorFactory;
     mNavigator = navigator;
     mGliderLoader = glideLoader;
     mPresentation = presentation;
-  }
 
-  @Override
-  public void visit(MessageText message) {
     initViewHolder();
-    setMessageDetails(message);
-    setTextContent(message.getText());
   }
 
   @Override
-  public void visit(MessageGeo message) {
-    initViewHolder();
-    setMessageDetails(message);
-    setTextContent(message.getGeoEntity().getText());
-    setGeoPanel(message.getGeoEntity().getGeometry(), message.getMessageId());
+  public void visit(TextFeature feature) {
+    setTextContent(feature.getText());
   }
 
   @Override
-  public void visit(MessageImage message) {
-    initViewHolder();
-    setMessageDetails(message);
-    setContent(message);
-    if (message instanceof MessageGeoImage) {
-      setGeoPanel(((MessageGeoImage) message).getGeoEntity().getGeometry(), message.getMessageId());
-    }
+  public void visit(GeoFeature feature) {
+    setGeoPanel(feature.getGeoEntity().getGeometry(), mPresentation.getMessage().getMessageId());
   }
 
   @Override
-  public void visit(MessageAlert message) {
-    initViewHolder();
-    setMessageDetails(message);
-    setContent(message.getAlert());
-    if (message.getAlert() instanceof GeoAlert) {
-      GeoAlert geoAlert = (GeoAlert) message.getAlert();
-      setGeoPanel(geoAlert.getEntity().getGeometry(), message.getMessageId());
-    }
+  public void visit(ImageFeature feature) {
+    setImageContent(feature);
   }
 
   @Override
-  public void visit(MessageUserLocation message) {
-    throw new RuntimeException("UserLocation messages should not be binded to whatsapp messages");
-  }
+  public void visit(AlertFeature feature) {
+    Alert alert = feature.getAlert();
+    String text = getAlertText(alert);
 
-  @Override
-  public void visit(MessageVectorLayer message) {
-    throw new RuntimeException("VectorLayer messages should not be binded to whatsapp messages");
+    if (!Strings.isNullOrEmpty(text)) {
+      setTextContent(text);
+    }
   }
 
-  @Override
-  public void visit(MessageSensor message) {
-    throw new RuntimeException("Sensor messages should not be binded to whatsapp messages");
+  private String getAlertText(Alert alert) {
+    if (alert.getType() == Alert.Type.VECTOR_LAYER) {
```

Comment:
I don't like methods like this.
What will happen if we have more than one type?
For every Enum we will create method?